core/src/main/java/com/adobe/aem/commons/assetshare/components/actions/download/impl/DownloadImpl.java

```diff
@@ -77,4 +102,22 @@ protected void init() {
     public String getZipFileName() {
         return StringUtils.removeEndIgnoreCase(zipFileName, ".zip");
     }
+
+    @Override
+    public boolean isMaxContentSize() {
+        if(maxContentSize != null && maxContentSize > 0 &&  maxContentSize < downloadContentSize) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public String getMaxContentSizeLimit() {
+        return UIHelper.getSizeLabel(maxContentSize, request);
```

Comment:
Noticed this on the PR for reading OSGi configs.. can the UIHelper get moved out from the Model to the HTL (is there a replacement for it perhaps?) This is squarely a presentation concern which belongs to the presentation layer (HTL) --  similar to how HTL supports date formatting. This really should just return a Long of bytes, if we can.

Reply:
no easy way to do this in HTL. Created two additional methods instead to explicitly return the label. The size limit and calculated size are returned as longs.